L1Trigger/DTTriggerPhase2/plugins/DTTrigPhase2Prod.cc

```diff
@@ -0,0 +1,719 @@
+          (*muonpath)->primitive(2)->laterality(),
+          (*muonpath)->primitive(3)->channelId(),
+          (*muonpath)->primitive(3)->tdcTimeStamp(),
+          (*muonpath)->primitive(3)->laterality(),
+          (*muonpath)->primitive(4)->channelId(),
+          (*muonpath)->primitive(4)->tdcTimeStamp(),
+          (*muonpath)->primitive(4)->laterality(),
+          (*muonpath)->primitive(5)->channelId(),
+          (*muonpath)->primitive(5)->tdcTimeStamp(),
+          (*muonpath)->primitive(5)->laterality(),
+          (*muonpath)->primitive(6)->channelId(),
+          (*muonpath)->primitive(6)->tdcTimeStamp(),
+          (*muonpath)->primitive(6)->laterality(),
+          (*muonpath)->primitive(7)->channelId(),
+          (*muonpath)->primitive(7)->tdcTimeStamp(),
+          (*muonpath)->primitive(7)->laterality(),
+      }));
+    }
+  }
+  filteredMetaPrimitives.clear();
+
+  if (debug_)
+    LogDebug("DTTrigPhase2Prod") << "DTp2 in event:" << iEvent.id().event() << " we found "
+                                 << correlatedMetaPrimitives.size() << " correlatedMetPrimitives (chamber)"
+                                 << std::endl;
+
+  if (dump_) {
+    LogDebug("DTTrigPhase2Prod") << "DTp2 in event:" << iEvent.id().event() << " we found "
+                                 << correlatedMetaPrimitives.size() << " correlatedMetPrimitives (chamber)"
+                                 << std::endl;
+
+    for (unsigned int i = 0; i < correlatedMetaPrimitives.size(); i++) {
+      LogDebug("DTTrigPhase2Prod") << iEvent.id().event() << " correlated mp " << i << ": ";
+      printmPC(correlatedMetaPrimitives.at(i));
+      LogDebug("DTTrigPhase2Prod") << "\n";
+    }
+  }
+
+  double shift_back = 0;
+  if (scenario_ == MC)  //scope for MC
+    shift_back = 400;
+  else if (scenario_ == DATA)  //scope for data
+    shift_back = 0;
+  else if (scenario_ == SLICE_TEST)  //scope for slice test
+    shift_back = 0;
+
+  // RPC integration
+  if (useRPC_) {
+    rpc_integrator_->initialise(iEventSetup, shift_back);
+    rpc_integrator_->prepareMetaPrimitives(rpcRecHits);
+    rpc_integrator_->matchWithDTAndUseRPCTime(correlatedMetaPrimitives);
+    rpc_integrator_->makeRPCOnlySegments();
+    rpc_integrator_->storeRPCSingleHits();
+    rpc_integrator_->removeRPCHitsUsed();
+  }
+
+  /// STORING RESULTs
+  vector<L1Phase2MuDTPhDigi> outP2Ph;
+
+  // Assigning index value
+  assignIndex(correlatedMetaPrimitives);
+  for (auto& metaPrimitiveIt : correlatedMetaPrimitives) {
+    DTChamberId chId(metaPrimitiveIt.rawId);
+    if (debug_)
+      LogDebug("DTTrigPhase2Prod") << "looping in final vector: SuperLayerId" << chId << " x=" << metaPrimitiveIt.x
+                                   << " quality=" << metaPrimitiveIt.quality
+                                   << " BX=" << round(metaPrimitiveIt.t0 / 25.) << " index=" << metaPrimitiveIt.index;
+
+    int sectorTP = chId.sector();
+    if (sectorTP == 13)
```

Comment:
many magic numbers

Reply:
this numbers are frequent in all the dt related code, added a comment so they are easly readable.